logback-classic/src/main/java/ch/qos/logback/classic/Logger.java

```diff
@@ -132,19 +132,14 @@ private boolean isRootLogger() {
     Logger getChildByName(final String childName) {
         if (childrenList == null) {
             return null;
-        } else {
-            int len = this.childrenList.size();
-            for (int i = 0; i < len; i++) {
-                final Logger childLogger_i = (Logger) childrenList.get(i);
-                final String childName_i = childLogger_i.getName();
-
-                if (childName.equals(childName_i)) {
-                    return childLogger_i;
-                }
+        }
+        for (final Logger child : childrenList) {
+            if (childName.equals(child.getName())) {
+                return child;
             }
```

Comment:
Use for-each to optimize

Reply:
I like this optimization, but I propose taking it further and ensuring that `childrenList` is always an instance of `CopyOnWriteArrayList`

```java
    transient private final List<Logger> childrenList = new CopyOnWriteArrayList<>();
```

or, with modifiers sorted

```java
    private final transient List<Logger> childrenList = new CopyOnWriteArrayList<>();
```

Then all the null checks could be removed and there would be not even a remote possibility that `childrenList`'s iterator could throw `ConcurrentModificationException`

Reply:
Very good suggestion, I can't agree more, updated.